uber/site_sections/registration.py

```diff
@@ -783,17 +783,15 @@ def feed(self, session, page='1', who='', what='', action=''):
             'who_opts': [who for [who] in session.query(Tracking).distinct().order_by(Tracking.who).values(Tracking.who)]
         }
 
-    def staffers(self, session, message='', order='first_name', search_text=''):
-        jobs, shifts, staffers = session.everything()
-        if search_text:
-            staffers = session.search(search_text, Attendee.staffing == True).options(joinedload(Attendee.shifts)).all()
+    def staffers(self, session, message='', order='first_name'):
+        staffers = (session.query(Attendee)
+                           .filter(Attendee.staffing == True, Attendee.badge_status.in_([c.NEW_STATUS, c.COMPLETED_STATUS]))
+                           .options(subqueryload(Attendee.shifts).subqueryload(Shift.job)).all())
```

Comment:
Here's the SQLAlchemy page on relationship loading strategies: http://docs.sqlalchemy.org/en/latest/orm/loading_relationships.html

To summarize ``subqueryload`` and what we're doing here, here's what it means when we say
``.options(subqueryload(Attendee.shifts).subqueryload(Shift.job))``
- Initially load the ``Attendee`` model objects but NOT any of its relationships.  So just loop over a single non-joined ``SELECT`` statement and load the results into a bunch of ``Attendee`` objects.
- The first time we say ``some_attendee.shifts``, issue a single ``SELECT`` statement for all of the ``Shift`` model objects for all of the Attendees we've selected.  That way we only need a single query for all the shifts we need instead of one for each Attendee.
- Similarly, the first time we say ``some_shift.job``, issue a single ``SELECT`` statement for all of the ``Job`` model objects for all of the Shifts for all of the Attendees we selected.

This is pretty sweet, and this sort of thing was one of the 3-4 reasons why I switched from Django to SQLAlchemy.  Although that's not very compelling considering that I made that switch and then COMPLETELY FAILED TO APPLY THIS KIND OF OPTIMIZATION FOR LITERALLY A YEAR AND A HALF AFTER MAKING THE SWITCH!!!!!

Ah well, better late than never :P